sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -99,3 +97,67 @@ class TypedAverage[IN](val f: IN => Double) extends Aggregator[IN, (Double, Long
     toColumn.asInstanceOf[TypedColumn[IN, java.lang.Double]]
   }
 }
+
+class TypedMinDouble[IN](val f: IN => Double) extends Aggregator[IN, Double, Double] {
+  override def zero: Double = Double.PositiveInfinity
```

Comment:
We can apply an aggregate on a dataset without groupBy. E.g.,

    val ds = Seq.empty[(Int, Int)].toDS
    ds.agg(typed.sum((x: (Int, Int))=> x._2)).show()

In this case, seems this typed `TypedMinDouble` will return the initial value from the `zero` variable. I've not tried it, but looks like it will get `Double.PositiveInfinity`. Beside the fact it's inconsistent with `aggregate.Min`, getting positive infinity here can't be a correct result too.